pkg/codegen/pcl/binder.go

```diff
@@ -320,6 +320,12 @@ func (b *binder) declareNodes(file *syntax.File) (hcl.Diagnostics, error) {
 					syntax: item,
 				}
 				declareDiags := b.declareNode(item.Labels[0], resource)
+
+				// If a resource uses an existing name, try adding a numeric suffix to make it unique.
```

Comment:
This should not be happening in the binder. This is a fix up the state importer should be doing when it generates ppl from state.